src/app/reports/reports.component.ts

```diff
@@ -0,0 +1,120 @@
+/** Angular Imports */
+import { Component, OnInit, ViewChild } from '@angular/core';
+import {MatPaginator, MatSort, MatTableDataSource} from '@angular/material';
+import { ActivatedRoute, Params, Router } from '@angular/router';
+
+/**
+ * Reports component.
+ */
+@Component({
+  selector: 'mifosx-app-reports',
+  templateUrl: './reports.component.html',
+  styleUrls: ['./reports.component.scss']
+})
+export class ReportsComponent implements OnInit {
+
+  filter: string;
+  /** Reports data. */
+  reportsData: any;
+  /** Columns to be displayed in reports table. */
+  displayedColumns: string[] = ['reportName', 'reportType', 'reportCategory'];
+  /** Data source for reports table. */
+  dataSource = new MatTableDataSource();
+
+  /** Paginator for reports table. */
+  @ViewChild(MatPaginator) paginator: MatPaginator;
+  /** Sorter for reports table. */
+  @ViewChild(MatSort) sort: MatSort;
+
+  /**
+   * Retrieves the reports data from `resolve`.
+   * @param {ActivatedRoute} route Activated Route.
+   * Prevents reuse of Route Params while navigating.
+   * @param {Router} router: Route.
+   */
+  constructor(private route: ActivatedRoute, private router: Router) {
+    this.router.routeReuseStrategy.shouldReuseRoute = () => false;
+
+    this.route.data.subscribe(( data: { reports: any }) => {
+      this.reportsData = data.reports;
+    });
+  }
+
+  ngOnInit() {
+    /*
+     *Fetches filter key from Route Params
+     */
+    this.route.params.subscribe((params: Params) => {
+      this.filter = params['filter'];
```

Comment:
Could this not be fetched in the constructor? And also follow the code pattern for fetching parameters from other components.